Format Currency Values with Commas in Contribution Flow

SudharakaP opened the pull request:
Related to https://github.com/opencollective/opencollective/issues/4471

![Peek 2021-07-26 09-12](https://user-images.githubusercontent.com/12435965/127022710-b9359de9-534e-4458-8b07-a7a963cb108c.gif)

Betree commented on lib/math.js:
```diff
@@ -1,37 +1,8 @@
+ * Adds commas to separate the digits of numbers (example: 100,000)
+ */
+export const formatCurrencyWithCommas = number => {
+  return number.toLocaleString('en-US');
```
We should somehow consider the locale of the user here. In French for example, `,` is the separator for decimals (`42,50€` => `€42.50`).

SudharakaP replied:
@Betree : Thanks, I've added some code to consider the browser locale so that if it's say french it's formatted that way. Here's a screenshot of how it looks. 🙂 

![image](https://user-images.githubusercontent.com/12435965/126934463-ea856173-6d04-48a3-96d2-363de6109866.png)

Let me know your thoughts.

Betree reviewed (commented):
Let's make this locale-friendly, numbers are not formatted the same way across languages so this might be confusing for users.

I also think it might be interesting to wait for some feedback in https://github.com/opencollective/opencollective/issues/4471. I would personally keep abbreviating when amount > `10000` (or any number that fits) as it's helpful for responsiveness.

SudharakaP commented:
> I also think it might be interesting to wait for some feedback in [opencollective/opencollective#4471](https://github.com/opencollective/opencollective/issues/4471). I would personally keep abbreviating when amount > `10000` (or any number that fits) as it's helpful for responsiveness.

Yes, we'll wait for feedback on the issue so that when it's finalized I'll add the abbreviation conditionally. 🙂

Betree commented on lib/math.js:
```diff
@@ -1,37 +1,12 @@
+export const formatCurrencyWithCommas = number => {
+  let locale = 'en-US';
+  if (typeof window !== 'undefined') {
+    locale = window.navigator.language;
```
That's not a great way to detect the locale, we should ideally rely on `react-intl`'s `intl.locale`. Maybe `formatCurrencyWithCommas` should take the locale as a parameter?

Also, isn't `toLocaleString` automatically using the browser locale if not provided?

SudharakaP replied:
@Betree : Sorry, I was confused that we have the detect the browser locale not the locale of open-collective. Anyways, I've just pushed a change to use the `intl.locale` instead and updated the above screen-cast as well. Let me know how it goes. 🙂

Betree commented on lib/math.js:
```diff
@@ -1,37 +1,12 @@
-    return round(number);
+ * Adds commas to separate the digits of numbers (example: 100,000)
+ */
+export const formatCurrencyWithCommas = number => {
```
The function name could be improved since the presence of commas depends on the language. With [many languages](https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html), it's actually the opposite: `100.000,00`

```suggestion
export const formatCurrencyWithSeparators = number => {
```

SudharakaP replied:
Done. 👍🏽

Betree commented on components/__tests__/Currency.test.js:
```diff
@@ -10,8 +10,8 @@ describe('Currency', () => {
-    snapshot(<Currency abbreviate value={1900000} currency="USD" />, { IntlProvider: { locale: 'fr' } });
+  it('currency format with separators', () => {
+    snapshot(<Currency formatWithSeparators value={4200} currency="USD" />);
+    snapshot(<Currency formatWithSeparators value={1900000} currency="USD" />, { IntlProvider: { locale: 'fr' } });
```
Would be great to add some test cases to show the variations across different locales

```suggestion
    snapshot(<Currency formatWithSeparators value={1900000} currency="USD" />, { IntlProvider: { locale: 'fr' } });
    snapshot(<Currency formatWithSeparators value={1900000} currency="USD" />, { IntlProvider: { locale: 'en-US' } });
```

SudharakaP replied:
Thanks. Done 👍🏽

Betree reviewed (changes requested):
What about other places using the `abbreviateAmount` option, like `TopContributors`?

SudharakaP commented:
> What about other places using the `abbreviateAmount` option, like `TopContributors`?

@Betree : I don't see anywhere in `TopContributors` where `abbreviateAmount` is used. Also `abbreviateAmount` isn't used anywhere, I think we should remove that prop entirely. WDYT? 🤔

Betree commented:
> > What about other places using the `abbreviateAmount` option, like `TopContributors`?
> 
> @Betree : I don't see anywhere in `TopContributors` where `abbreviateAmount` is used. Also [`abbreviateAmount`](https://github.com/opencollective/opencollective-frontend/blob/4590f4e770f18ee98f6c3f7d488da3f099111be1/components/FormattedMoneyAmount.js#L83) isn't used anywhere, I think we should remove that prop entirely. WDYT? thinking

There: https://github.com/opencollective/opencollective-frontend/blob/3ca5f57fa9b1c2c08ff59242e6f34faf1a21f471/components/collective-page/TopContributors.js#L133

From what I see the prop is already removed from `FormattedMoneyAmount` in this PR, but we should review all places using this prop to remove it as well.

SudharakaP commented:
> From what I see the prop is already removed from `FormattedMoneyAmount` in this PR, but we should review all places using this prop to remove it as well.

@Betree : Ops, sorry was a bit confused there. But yes I've removed that remaining prop as we've removed `abbreviateAmount`. Let me know if all looks good now. 👍🏽

Betree reviewed (approved):
Looks good, we can try that and iterate if needed
